docs/libraries/mst-form-type.md

```diff
@@ -0,0 +1,95 @@
+# MST Form Type
+
+npm: https://www.npmjs.com/package/mst-form-type
+
+## Introduction
+
+Librarys like Ant Design has a build-in Form component that can hold fields status and validation rules. This component houses its own store to sustain internally related field statuses and can be conveniently used independently. integrating Form component with MobX State Tree models can pose significant challenges as business logic grows intricately complex.
+
+The `mst-form-type` library comes into play. It provides the field management feature of Ant Design Form component in Mobx State Tree model way. Users can only use the UI of a component library, but keep logic inside your Mobx State Tree model context, instead of syncing status changes between Form component and model.
+
+Please note: The `mst-form-type` library primarily provides model types for the form structure. It does not encompass business logic related to field interactions.
+
+## Setup
+
+Setting up and utilizing the `mst-form-type` library is straightforward:
+
+1. Install the `mst-form-type` library via npm:
+
+```sh
+npm install mst-form-type
+```
+
+2. Ensure you have `mobx-state-tree ^5.0.0` installed:
+
+```sh
+npm install mobx-state-tree@^5.0.0
+```
+
+3. Declare a form schema and create the form model, subsequently, all the relevant field models would then be created in a form model. An example schema look like this:
+
+```typescript
```

Comment:
**suggestion:** can you show a full example, including all the imports, this schema definition, and then how the library would use it? I see something like that on npm:

```js

// render dynamic fields
model.dynamicForm['price'].fields.map(fields => ({ ... }))

// field action
model.dynamicForm.onAdd('price') // this will use field default value in schema
model.dynamicForm.onRemove('price', fields.id)

// form action
model.dynamicForm.submit()
model.dynamicForm.reset()
model.dynamicForm.clear('price') // this will clear all dynamic fields, including default ones
```